streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1613,7 +1640,14 @@ private List<Task> standbyTaskIterable() {
     }
 
     private Stream<Task> standbyTaskStream() {
-        return tasks.allTasks().stream().filter(t -> !t.isActive());
+        if (stateUpdater != null) {
+            return Stream.concat(
+                stateUpdater.getStandbyTasks().stream(),
+                tasks.allTasks().stream().filter(t -> !t.isActive())
+            );
+        } else {
```

Comment:
With enabled state updater, the state updater owns standby tasks. However, when the state updater is shut down it moves its stand by tasks to the stream thread for closing out. Thus, we also need to add `tasks.allTasks().stream().filter(t -> !t.isActive())` here.